Tests/SwiftParserTest/ExpressionTests.swift

```diff
@@ -2079,32 +2161,141 @@ final class StatementExpressionTests: XCTestCase {
+      (def)4️⃣"5️⃣
+      """#,
+      diagnostics: [
+        DiagnosticSpec(
+          locationMarker: "2️⃣",
+          message: "invalid escape sequence in literal"
+        ),
+        DiagnosticSpec(
+          locationMarker: "3️⃣",
+          message: #"expected '"' to end string literal"#,
+          notes: [NoteSpec(locationMarker: "1️⃣", message: #"to match this opening '"'"#)],
+          fixIts: [#"insert '"'"#]
+        ),
+        DiagnosticSpec(
+          locationMarker: "4️⃣",
+          message: "consecutive statements on a line must be separated by newline or ';'",
+          fixIts: ["insert newline", "insert ';'"]
+        ),
+        DiagnosticSpec(
+          locationMarker: "5️⃣",
+          message: #"expected '"' to end string literal"#,
+          notes: [NoteSpec(locationMarker: "4️⃣", message: #"to match this opening '"'"#)],
+          fixIts: [#"insert '"'"#]
+        ),
+      ],
+      applyFixIts: [#"insert '"'"#, "insert ';'"],
       fixedSource: #"""
         "abc\"
         (def); ""
         """#
     )
   }
 
-  func testUnterminatedString7() {
+  func testUnterminatedString7a() {
+    assertParse(
+      #"""
+      #1️⃣
+      "abc"2️⃣#3️⃣
+      """#,
+      diagnostics: [
+        DiagnosticSpec(
+          locationMarker: "1️⃣",
+          message: "expected identifier in macro expansion",
+          fixIts: ["insert identifier"]
+        ),
+        DiagnosticSpec(
+          locationMarker: "2️⃣",
+          message: "consecutive statements on a line must be separated by newline or ';'",
+          fixIts: ["insert newline", "insert ';'"]
+        ),
+        DiagnosticSpec(
+          locationMarker: "3️⃣",
+          message: "expected identifier in macro expansion",
+          fixIts: ["insert identifier"]
+        ),
+      ],
+      applyFixIts: ["insert identifier", "insert newline"],
+      fixedSource: #"""
+        #<#identifier#>
+        "abc"
+        #<#identifier#>
+        """#
+    )
+  }
+
+  func testUnterminatedString7b() {
     assertParse(
       #"""
       #1️⃣
       "abc"2️⃣#3️⃣
       """#,
       diagnostics: [
         DiagnosticSpec(locationMarker: "1️⃣", message: "expected identifier in macro expansion", fixIts: ["insert identifier"]),
-        DiagnosticSpec(locationMarker: "2️⃣", message: "consecutive statements on a line must be separated by ';'", fixIts: ["insert ';'"]),
+        DiagnosticSpec(
+          locationMarker: "2️⃣",
+          message: "consecutive statements on a line must be separated by newline or ';'",
+          fixIts: ["insert newline", "insert ';'"]
+        ),
         DiagnosticSpec(locationMarker: "3️⃣", message: "expected identifier in macro expansion", fixIts: ["insert identifier"]),
       ],
+      applyFixIts: ["insert identifier", "insert ';'"],
       fixedSource: #"""
         #<#identifier#>
         "abc"; #<#identifier#>
         """#
     )
   }
 
-  func testUnterminatedString8() {
+  func testUnterminatedString8a() {
```

Comment:
I’m not sure if duplicating all these test cases for insertion of `;` and a newline provides a huge amount of value. But it doesn’t hurt either, so all good.